File: JavaScript/Maintainability Nesting Depth/workspace/js_nesting_benchmark/cases/loop_nesting.js
export function loopNesting(items) {
  let total = 0;
  for (const item of items) {
    if (item > 0) {
      for (const nested of items) {
        if (nested > 10) {
          for (const deep of items) {
            if (deep > 20) {
              total += deep;
            }
          }
        }
      }
    }
  }
  return total;
}

File: CSharp/Maintainability Nesting Depth/workspace/cs_nesting_benchmark/cases/LoopNesting.cs
namespace Cases;

public class LoopNesting
{
    public int NestedLoops(int[] values)
    {
        var total = 0;
        foreach (var item in values)
        {
            if (item > 0)
            {
                for (var i = 0; i < values.Length; i++)
                {
                    if (values[i] > 10)
                    {
                        for (var j = 0; j < values.Length; j++)
                        {
                            if (values[j] > 20)
                            {
                                total += values[j];
                            }
                        }
                    }
                }
            }
        }

        return total;
    }
}
